### PR TITLE
feat: Add TypeScript module

### LINKED ISSUE #210: Add TypeScript module
## Feature Request
Add a [TypeScript](https://www.typescriptlang.org/) module to the prompt when the user is in a typescript project

#### Describe the solution you'd like

I want the prompt to include a module with typescript's version when I'm in a typescript project

### PR DESCRIPTION
**Pull request by oransimhony**
Added a typescript module to the prompt that shows the current TypeScript version when the user is inside a TypeScript project.

#### Description

Added a new module based on the other ones that calls `tsc` (TypeScript's compiler).
Added tests to make sure that the module works correctly.
Added integration tests and updated the `Dockerfile`.
Updated the documentation to contain info about the typescript module.

#### Motivation and Context

Closes #210 

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):
Screen Shot 2019-08-20 at 19 45 33

#### How Has This Been Tested?

- [x] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/modules/typescript.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        .set_extensions(&["ts"])
+        .scan();
+
+    if !is_ts_project {
+        return None;
+    }
+
+    match get_typescript_version() {
+        Some(typescript_version) => {
+            const TYPESCRIPT_CHAR: &str = "■ ";
```
Comment (youssefhabri) [inline src/modules/typescript.rs]: You can probably use the 🇹 emoji, can't you? :thinking:

Comment (oransimhony) [reply src/modules/typescript.rs]: I've tried it and this is how it looks.
Screen Shot 2019-08-20 at 20 50 22
If that's better in your opinion I can commit the changes 😄

Comment (youssefhabri) [reply src/modules/typescript.rs]: aah, I guess font support might be an issue. I'm not sure which to choose in that case, maybe @matchai can decide?

Comment (johnletey) [reply src/modules/typescript.rs]: @youssefhabri I like the square better

**Comment by matchai** [comment]
Closing this PR for the time being.
It can be reopened if we decide to revisit the TypeScript module. 🙂

**Pull request closed without merging**